### DISCUSSION TITLE
what is a shortcut command to select to enclosing bracket?

### DISCUSSION THREAD

**Discussion by tonku321** [zed-industries/zed#22208] [Feature Requests]
i tried using`editor::MoveToEnclosingBracket` in vim visual mode, but it drops the selection :(

**Comment by 0x2CA** [comment]
I don't quite understand the effect you're going for, but I've made some guesses

Zed has already supported common TextObject.

```json
 {
    "context": "vim_operator == a || vim_operator == i || vim_operator == cs",
    "bindings": {
      "w": "vim::Word",
      "shift-w": ["vim::Word", { "ignorePunctuation": true }],
      "t": "vim::Tag",
      "s": "vim::Sentence",
      "p": "vim::Paragraph",
      "'": "vim::Quotes",
      "`": "vim::BackQuotes",
      "\"": "vim::DoubleQuotes",
      "|": "vim::VerticalBars",
      "(": "vim::Parentheses",
      ")": "vim::Parentheses",
      "b": "vim::Parentheses",
      "[": "vim::SquareBrackets",
      "]": "vim::SquareBrackets",
      "r": "vim::SquareBrackets",
      "{": "vim::CurlyBrackets",
      "}": "vim::CurlyBrackets",
      "shift-b": "vim::CurlyBrackets",
      "<": "vim::AngleBrackets",
      ">": "vim::AngleBrackets",
      "a": "vim::Argument",
      "i": "vim::IndentObj",
      "shift-i": ["vim::IndentObj", { "includeBelow": true }],
      "f": "vim::Method",
      "c": "vim::Class"
    }
  },
```

If you want to select a curly brackets, you can use `va{`
If you just want to select the text to the next}, you can use the search or find
such as `vf}` or `v/}enter`

Of course the lookup defaults to the current line only, you can turn on the multi-line setting

```json
  "vim": {
    "use_multiline_find": true,
  },
```

So you can use `n/N` or `;/,` to expand

**Comment by tonku321** [comment]
thanks to @helgemahrt, `vim::Matching` is what i was looking for, since it works the same as `editor::MoveToEnclosingBracket`, except it also does not drop the selection in visual.
i would expect to see something like `editor::SelectToEnclosingBracket` though :)

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
